state-reconstruct-fetcher/src/types/v3.rs

```diff
@@ -141,9 +148,113 @@ fn parse_pubdata_from_calldata(
 
     // Parse compressed state diffs.
     // NOTE: Is this correct? Ignoring the last 32 bytes?
-    let end_point = bytes.len() - 32;
-    let mut state_diffs = parse_compressed_state_diffs(&bytes[..end_point], pointer)?;
+    let diff_bytes = if shorten {
+        let end_point = bytes.len() - 32;
+        &bytes[..end_point]
+    } else {
+        bytes
+    };
```

Comment:
We _are_ in v3 here - it calls the function with both true and false...

Reply:
Gotcha! I don't see a reason why we need an explicit `parse_pubdata_from_calldata` then. I believe that we should be able to reuse the same one as `V2` (passing in `bytes[..len-32] of course` when we're just using calldata).

Reply:
Well, maybe we don't, but since it was already there, I deferred to your judgement... :-)

Reply:
...but OK, I've combined the functions - it isn't _quite_ a compatible change, since the longer slice could potentially be processed differently before it was shortened, but that was probably an error anyway...